single log line

Seetaramayya opened the pull request.

Seetaramayya committed:
Implemented review comments

Seetaramayya committed:
implemented review comments

Seetaramayya committed:
Removed one more type def

jordi133 commented:
Approved for merge after changing type alias Monitor to AlertRuleActorRef and consistent use of sender()

Seetaramayya committed:
Removed `monitor` references

Seetaramayya committed:
Merge branch 'develop' of https://github.com/scala-academy/performance-analysis into feature/#16-single-log-line

# Conflicts:
#	src/test/scala/performanceanalysis/logreceiver/LogReceiverSpec.scala

Seetaramayya committed:
Resolved merge conflict

ruud commented on src/it/scala/performanceanalysis/LogSubmissionTest.scala:
```diff
@@ -14,18 +17,42 @@ class LogSubmissionTest extends IntegrationTestBase with ScalaFutures with Twitt
-        assert(result.getStatusCode() === 405)
-        Then("the response should have statuscode 405")
+      whenReady(response, timeout(1 second)) { result =>
+        result.getStatusCode() shouldBe 405
```
for consistency:
405 should be changed into StatusCodes.MethodNotAllowed

Seetaramayya replied:
I agree

ruud commented on src/it/scala/performanceanalysis/LogSubmissionTest.scala:
```diff
@@ -14,18 +17,42 @@ class LogSubmissionTest extends IntegrationTestBase with ScalaFutures with Twitt
-        Then("the response should have statuscode 405")
+      whenReady(response, timeout(1 second)) { result =>
+        result.getStatusCode() shouldBe 405
+        Then("the response should have status code 405")
```
for consistency:
405 should be changed into StatusCodes.MethodNotAllowed

Seetaramayya replied:
I agree, I'll change

ruud commented on src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala:
```diff
@@ -13,28 +13,31 @@ object LogReceiverActor {
-      handleNewLogParser(logParserActors, componentName, newLogParser)
+  private type ComponentId = String
+  private type LogParser = ActorRef
+  private type LogParsers = Map[ComponentId, LogParser]
```
As before, maybe its not a good idea to create a type for a Map. The name suggests a List or Set.

Seetaramayya replied:
I'll change

jordi133 replied:
Agree, `LogParsersById` is a bit more clear (and prevents further discussion on this ;)

Seetaramayya replied:
:+1:

ruud replied:
Not creating a new type would be consistent with the rest

ruud commented on src/main/scala/performanceanalysis/server/Protocol.scala:
```diff
@@ -60,7 +65,7 @@ object Protocol {
 
   /**
-    * Used by Administrator towards LogReceiver to notify it of a new LogReceiver actor
+    * Used by Administrator towards LogReceiver to notify it ofa new LogReceiver actor
```
of a is ok, isn't it ;)

Seetaramayya replied:
Indeed, I think `intellij` did it. I'll change.

ruud commented:
Just some minor comments from me.
+1 for merging pull-request

Seetaramayya committed:
For consistency modified

Seetaramayya committed:
Corrected typo

ruud committed:
First integration test

Given the server is running and I registered component
"logsObtainableComp"
And I posted a logline "some action took 101 seconds" on this component
When I do a GET to /components/logsObtainableComp/logs
Then the result should have statuscode 200
And the content should contain "some action took 101 seconds"

ruud committed:
revert changes

Sorry

Seetaramayya commented:
@ruud Did you push changes by mistake? Or Am I missing the point?

Seetaramayya committed:
To trigger build
